### PR TITLE
Refs #16740 - Update all template to use params macros

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

File: test/test_helper.rb [hunk 1]
```diff
@@ -50,6 +50,22 @@ def param_true?(name)
   def param_false?(name)
```
Comment (stbenjam) [inline test/test_helper.rb]: Should this get removed?

Comment (ares) [reply test/test_helper.rb]: yeah, good idea, this will check for deprecated use of it in new templates

**Comment by ares** [comment]
I kept the changes in separate commits for easier reviewing, let me know if you want me to squash it.

**Comment by mmoll** [comment]
@ares please rebase.

**Comment by ares** [comment]
rebased

**Comment by mmoll** [comment]
`@host.param_false?` and `@host.param_true?` are still used in snippets/freeipa_register.erb, looking fine otherwise.

**Comment by ares** [comment]
Updated, but I guess @dLobatog wanted to do the revert of the original change https://groups.google.com/forum/#!topic/foreman-dev/P364HozAYZA, I think it's the time to decide.

**Comment by mmoll** [comment]
this needs a rebase anyway now ;)

**Comment by ares** [comment]
git++ rebased

**Comment by ekohl** [comment]
What would be an actual improvement and a reason to use this macro: allow it to specify a default if unset. So instead of
```partitioning_method = host_param('partitioning-method') ? host_param('partitioning-method') : 'regular'```
you could use
```partitioning_method = host_param('partitioning-method', 'regular')```

Other than that it currently has conflicts again.

**Comment by ares** [comment]
I don't want to spend much time on this until https://github.com/theforeman/foreman/pull/4219 gets clear. I'm happy to improve macros in this way if they remains in 1.15 codebase.

**Comment by mmoll** [comment]
as this will stay as-is in core, this needs a rebase again :)

**Comment by ares** [comment]
yup, on my list with also providing the migration again, I hope this or next week

**Comment by ares** [comment]
> What would be an actual improvement and a reason to use this macro: allow it to specify a default if unset. 

I'm going to reopen https://github.com/theforeman/foreman/pull/4187 so I'll incorporate this in.

@mmoll rebased

**Comment by mmoll** [comment]
merged, díky @ares!

**Pull request merged**